lib/transaction/decodeAndDeserialize.ts

```diff
@@ -0,0 +1,12 @@
+import * as bs58 from "bs58";
+import { VersionedMessage, VersionedTransaction } from "@solana/web3.js";
+
+export function decodeAndDeserialize(tx: string): VersionedMessage {
+  try {
+    const messageBytes = bs58.default.decode(tx);
+    return VersionedMessage.deserialize(messageBytes);
```

Comment:
The "issue" is that here we rely on the fact its only a versioned tx

Reply:
updated in [728c8e2](https://github.com/Squads-Protocol/squads-v4-public-ui/pull/11/commits/728c8e223eb16ef320267100b78c3a5330ffc379) to handle both types